src/views.js

```diff
@@ -438,13 +438,23 @@ const AttachmentChatboxMessageView = new Lang.Class({
         let thumbnailFactory = Thumbnailer.forSize(GnomeDesktop.DesktopThumbnailSize.LARGE);
         let preview = getPreviewForFile(this.state.path, thumbnailFactory);
         if (preview.thumbnail) {
+            this.attachment_contents.orientation = Gtk.Orientation.VERTICAL;
+
             // Where we are using a thumbnail, use GtkCSSProivder to
             // set the background image
+            let naturalWidth = this.get_preferred_size()[1].width;
+            let minWidth = preview.thumbnail.width < naturalWidth ? naturalWidth : preview.thumbnail.width;
```

Comment:
I'm still not quite sure I understand here. We need to do the following:

 1. Check which width to use (might either be natural width or thumbnail width, depending on which is larger)
 2. Compute height by dividing width by aspect ratio.
 3 .Specifying width and height as appropriate.

In that sense the natural width comparison is important for making this work. I suspect what might have made this confusing is:

 > I think that this check will always end up with preview.thumbnail.width, because this could not possibly have a larger size, as it will only be sized afterwards

In this case, `get_preferred_size` can (and is, in some cases) actually be larger than the thumbnail itself. This is because we are checking on the *container* (which also contains the title and description).